extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -25,11 +190,35 @@ export let KeyedShoppingConfigDef;
  */
 export function getShoppingConfig(element) {
   return getElementConfig(element).then((config) => {
-    //TODO(#36412): Add call to validate config here.
+    const shoppingTagIndicesToRemove = [];
+    let currentShoppingTagIndex = 0;
+    const areConfigsValid = config['items'].reduce((item1, item2) => {
+      isValidConfigSection_ = true;
+      let isValidConfig = validateConfig(item2);
+      isValidConfig &&= isValidConfigSection_;
+      if (!isValidConfig) {
+        shoppingTagIndicesToRemove.push(currentShoppingTagIndex);
+      }
+      currentShoppingTagIndex++;
+      return item1 && isValidConfig;
+    }, true);
+
+    if (!areConfigsValid) {
+      user().warn(
+        'AMP-STORY-SHOPPING-CONFIG',
+        `Required fields are missing. Please add them in the shopping config. See the error messages above for more details.`
+      );
+      let indexOffset = 0;
+      for (const index of shoppingTagIndicesToRemove) {
+        //need to keep track of accumulated indices to make sure that the splice index is correct!
+        config['items'].splice(index + indexOffset, 1);
+        indexOffset--;
+      }
+    }
+
     return keyByProductTagId(config);
```

Comment:
I think we can greatly simplify this logic by using `filter()` instead of `reduce()`:
```
const allItems = config[‘items’];
const validItems = allItems.filter(item => validateConfig(item));
if (allItems.length != validItems.length) {
  user().warn();
}
return keyByProductTagId(validItems);
```
To make this work, you’ll also need to update `keyByProductTagId()` to take an items parameter.

Reply:
I agree, except we still have the problem with the `createValidateConfigArray` and `createValidateConfigObject` object's boolean values not being taken in to consideration if we do it this way - see https://github.com/ampproject/amphtml/pull/37474#discussion_r819069094, and those are important parts of blocking validation. If you can find a workaround that includes taking the validation results from the above two functions into account, then I'm happy to make the switch!